### PR TITLE
Fix for issue with reverting the datasource to default when sys admin updates an org

### PR DESCRIPTION
**Pull request by alvak1**
The fix resolves the issue of datasource reverting to default value when a change is made to the organisation

### CODE REVIEW

File: site/pages/sysadmin/org.page.tsx [hunk 1]
```diff
@@ -160,12 +161,16 @@ export const getServerSideProps = async (
...
-        "ID",
-        !errorCookie && orgId && orgInfo ? orgInfo : undefined,
-    );
+    let pageState: PageState<Partial<Organisation>> = {
+        errors: [],
+        inputs: {},
+    };
+
+    if (errorCookie) {
+        pageState = JSON.parse(errorCookie) as PageState<Partial<Organisation>>;
```
Comment (alainajaffer) [inline site/pages/sysadmin/org.page.tsx:170]: could we not use zod to parse this?

File: site/pages/sysadmin/org.page.tsx [hunk 2]
```diff
@@ -117,6 +117,7 @@ const ManageOrgs = (props: PageState<Partial<ManageOrgProps>>): ReactElement =>
 
                 <input type="hidden" name="PK" value={pageState.inputs.PK} />
                 <input type="hidden" name="adminAreaCodes" value={pageState.inputs.adminAreaCodes ?? []} />
+                <input type="hidden" name="mode" value={JSON.stringify(pageState.inputs.mode)} />
```
Comment (alainajaffer) [inline site/pages/sysadmin/org.page.tsx:120]: will mode always be defined otherwise this would fail?

File: site/pages/sysadmin/org.page.tsx (hunk 1 above)
Comment (alvak1) [reply site/pages/sysadmin/org.page.tsx:170]: As the errorCookie is json string we can use zod to parse it

File: site/pages/sysadmin/org.page.tsx (hunk 2 above)
Comment (alvak1) [reply site/pages/sysadmin/org.page.tsx:120]: yes it will always be returned because the org will always have this info in db

**Review by alainajaffer — APPROVED** [review]

**Pull request merged**